src/shared/sleep-config.c

```diff
@@ -208,66 +325,203 @@ static bool battery_discharge_rate_is_valid(int battery_discharge_rate) {
 
 /* Battery percentage discharge rate per hour is read from specific file. It is stored along with system
  * and battery identifier hash to maintain the integrity of discharge rate value */
-int get_battery_discharge_rate(void) {
-        _cleanup_free_ char *hash_id_discharge_rate = NULL, *stored_hash_id = NULL, *stored_discharge_rate = NULL;
-        const char *p;
+static int get_battery_discharge_rate(sd_device *dev, int *ret) {
+        _cleanup_free_ char *stored_hash_id = NULL, *stored_discharge_rate = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
         uint64_t current_hash_id, hash_id;
+        const char *p;
         int discharge_rate, r;
```

Comment:
Please move `stored_hash_id`, `stored_discharge_rate`, `hash_id`, `discharge_rate` in the loop below.